test/e2e/framework/job/wait.go

```diff
@@ -115,24 +120,25 @@ func WaitForJobFailed(c clientset.Interface, ns, jobName string) error {
 	})
 }
 
-// waitForJobCondition waits for the specified Job to have the expected condition with the specific reason.
-func WaitForJobCondition(ctx context.Context, c clientset.Interface, ns, jobName string, cType batchv1.JobConditionType, reason string) error {
+// WaitForJobCondition waits for the specified Job to have the expected condition with the specific reason.
+// When the nil reason is passed, this never care the differences between the expected and actual condition reason.
```

Comment:
```suggestion
// When the nil reason is passed, the "reason" string in the condition is
// not checked.
```

Reply:
I addressed this in https://github.com/kubernetes/kubernetes/compare/b6cf133acea9f2bc530921cc156f9d5a399af24f..0d4f18bd5b8e05211e985f31781cced9aac42e78.